lib/kennel/syncer.rb

```diff
@@ -55,8 +70,11 @@ def update
         message = "#{klass.api_resource} #{a.fetch(:tracking_id)} #{id}"
         Kennel.out.puts "Deleting #{message}"
         @api.delete klass.api_resource, id
+        update_log << [:delete, klass.api_resource, id]
         Kennel.out.puts "#{LINE_UP}Deleted #{message}"
       end
+
+      Update.new(update_log: update_log)
```

Comment:
Note that `update_log` exposes quite a bit of data (and all undocumented). For the purposes of the corresponding zendesk kennel PR we _could_ have made it expose just the tracking ids.

Reply:
how is the update-log different from a `Plan` ? ... it has the ids resolved ?
can we use the `Plan` internally and then resolve the ids in the plan to have the same data structure at the end ?

Reply:
Well.... at the moment, perhaps not much. My longer term view however is this:

The way that `plan` works at the moment – and therefore what the plan actually _is_ – is very simplistic. I prefer to think of it not so much of a plan, but more of a diff; a comparison. If we compare A to B, then this is added, this is changed, this is removed.

If the job then is resolve those differences (assuming it's a non-empty set), then we need to _do_ something. What, exactly? Well, that's the plan. Or, once performed, the results.

Our methodology is already showing its shortcomings. If we for example create a monitor M with tracking id K and type T, and then we update kennel so that the type now needs to be T', then (IIRC) Kennel currently can't handle that, because Datadog does not allow a monitor's type to be changed. However, it is solvable: instead of updating the monitor, delete monitor M then create monitor M' type T' (and with the same tracking ID K). In this case, the _diff_ would be "Monitor K change T => T'", but the _plan_ would be "delete M and create M'".

Another example: circular references. The reason that I haven't yet merged https://github.com/grosser/kennel/pull/160 is that it would allow for / increase the likelihood of circular references. Monitor M, whose message text includes a link to troubleshooting dashboard D. But dashboard D contains a view of monitor M. So if M => D and D => M, and neither exist, so the diff is "both D and M are new", what's the plan? Possible answer: create M (but with the message text _only_ containing the kennel tracking text); create D, referring to M; update M to set its message text correctly, linking to D.

To return to your question, "what's the difference between the plan and the update log?". What we currently call `plan` is a diff. What I'm calling update_log is closer to a plan – but retrospective.